stats/opentelemetry/e2e_test.go

```diff
@@ -582,3 +601,365 @@ func pollForWantMetrics(ctx context.Context, t *testing.T, reader *metric.Manual
 
 	return fmt.Errorf("error waiting for metrics %v: %v", wantMetrics, ctx.Err())
 }
+
+// spanInformation is the information received about the span. This is a subset
+// of information that is important to verify that gRPC has knobs over, which
+// goes through a stable OpenTelemetry API with well-defined behavior. This keeps
+// the robustness of assertions over time.
+type spanInformation struct {
```

Comment:
should be traceSpanInformation? Also, better to put all structs and helper methods together at the top